sound/soc/soc-topology.c

```diff
@@ -1268,10 +1268,12 @@ static int soc_tplg_dapm_graph_elems_load(struct soc_tplg *tplg,
 		snd_soc_dapm_add_routes(dapm, routes[i], 1);
 	}
 
-	/* free memory allocated for all dapm routes in case of error */
-	if (ret < 0)
-		for (i = 0; i < count ; i++)
-			kfree(routes[i]);
+	/* free memory allocated for all dapm routes not added to the list in case of error */
+	if (ret < 0) {
+		i++;
+		while (i < count)
```

Comment:
@plbossart Not sure I understand the initial "i++".  If e.g. strnlen() check fails, routes[i] is not added to dapm_add_routes and if you i++, it's not freed.

If I missed something and i++ is needed, then as @lyakh is ooo, it falls on me to request this to be a for loop. ;)

Reply:
You're right, I was only trying to deal with the last 'break' in the previous loop happens after the 'i' route was added to the loop. So you need to increment to deal with the rest i+1..count-1.
I should move the i++ before the break, that should cover all cases.

Reply:
fixed now